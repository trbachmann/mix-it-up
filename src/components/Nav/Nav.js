import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

export const Nav = (props) => {
  return(
    <div className='Nav--div'>
      <NavLink className='Nav--NavLink--margin' to='/'>Home</NavLink>
      <NavLink className='Nav--NavLink' to='/my-recipes'>My {props.totalUserRecipes} Recipes</NavLink>
    </div>
  );
}

Nav.propTypes = {
  totalUserRecipes: PropTypes.number
}