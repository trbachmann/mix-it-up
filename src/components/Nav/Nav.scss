  .Nav--div {
    background-color: #fff;
    height: 3em;
    display: flex;
    align-items: center;
    margin-bottom: 2em;
  } 

 .Nav--NavLink {
   font-size: 1.5em;
   border-bottom: 2px solid transparent;
   text-decoration: none;
   color: black;
   margin-right: 2em;
   padding: .4em 0;
   &:hover {
     border-bottom: 2px solid $darkMaroon;
   }
 }

 .Nav--NavLink--margin {
   @extend .Nav--NavLink;
   margin-left: auto;
 }

