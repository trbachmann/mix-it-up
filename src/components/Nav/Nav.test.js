import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';

const mocktotalUserRecipes = 4;

describe('Nav', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Nav totalUserRecipes={mocktotalUserRecipes}/>);
    expect(wrapper).toMatchSnapshot();
  })
});