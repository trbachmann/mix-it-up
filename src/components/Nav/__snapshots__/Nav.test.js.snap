// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<div
  className="Nav--div"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="Nav--NavLink--margin"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="Nav--NavLink"
    to="/my-recipes"
  >
    My 
    4
     Recipes
  </NavLink>
</div>
`;
